experimental/PlaneCurve/ProjEllipticCurve.jl

```diff
@@ -0,0 +1,459 @@
+   return P.Hecke_Pt
+end
+
+################################################################################
+
+@doc Markdown.doc"""
+    curve(P::Point_EllCurve)
+
+Return the curve on which the point `P` is considered.
+"""
+function curve(P::Point_EllCurve)
+   return P.C
+end
+
+################################################################################
+# Helping function
+
+function proj_space(P::Point_EllCurve)
+   return P.Pt.parent
+end
+
+################################################################################
+
+function ==(P::Point_EllCurve, Q::Point_EllCurve)
+   return P.C == Q.C && P.Pt == Q.Pt
+end
+
+################################################################################
+function Base.:+(P::Point_EllCurve, Q::Point_EllCurve)
+   E = P.C
+   PP = proj_space(P)
+   A = _EllCrvPt(P) + _EllCrvPt(Q)
+   return _point_fromweierstrass(E, PP, A)
+end
+
+################################################################################
+
+function Base.:-(P::Point_EllCurve)
+   E = P.C
+   PP = proj_space(P)
+   return _point_fromweierstrass(E, PP, -_EllCrvPt(P))
+end
+
+function Base.:-(P::Point_EllCurve, Q::Point_EllCurve)
+   E = P.C
+   PP = proj_space(P)
+   A = _EllCrvPt(P) + (-_EllCrvPt(Q))
+   return _point_fromweierstrass(E, PP, A)
+end
+################################################################################
+
+function Base.:*(n::Int64, P::Point_EllCurve)
+   E = curve(P)
+   PP = proj_space(P)
+   return _point_fromweierstrass(E, PP, n*_EllCrvPt(P))
+end
+
+################################################################################
+@doc Markdown.doc"""
+    order(E::ProjEllipticCurve)
+Given an elliptic curve `E` over a finite field $\mathbf F$, computes
+$\#E(\mathbf F)$.
+"""
+function Oscar.order(E::ProjEllipticCurve)
+   return Hecke.order(E.Hecke_ec)
+end
+
+################################################################################
+# helping function
+
+function contract(i::Int, F::Oscar.MPolyElem{S}) where {S <: FieldElem}
+   R = parent(F)
+   P = MPolyBuildCtx(R)
+   L = exponent_vectors(F)
+   e = zeros(Int64, nvars(R)-1)
+   insert!(e, i, -1)
+   A = [v + e for v in L]
+   Co = coefficients(F)
+   B = [c for c in Co]
+   C = []
+   E = []
+   j = 0
+   for v in A
+      j = j + 1
+      if v[i] >= 0
+         push!(C, B[j])
+         push!(E, v)
+      end
+   end
+   Z = zip(C, E)
+   for (c, v) in Z
+      push_term!(P, c, v)
+   end
+   return finish(P)
+end
+
+################################################################################
+# helping function
+
+function _eva(F::Oscar.MPolyElem{S}) where {S <: FieldElem}
+   R = parent(F)
+   nvars(R) == 3 || error("wrong number of variables")
+   X = gens(R)
+   return evaluate(evaluate(F, [X[1], X[2], R(0)]), [R(1), X[1], R(1)])
+end
+
+
+
+################################################################################
+# This code is copied from Macaulay2, based on Tibouchi, M. ''A Nagell Algorithm
```

Comment:
```suggestion
# This code is inspired from Macaulay2, based on Tibouchi, M. ''A Nagell Algorithm
```
Otherwise we would have to add some copyright notices?